### DISCUSSION TITLE
Manual Refill??

### DISCUSSION THREAD

**Discussion by afm497** [vladimir-bukhtoyarov/bucket4j#303] [Q&A]
Is it possible to configure a bucket such that it could operate in a mode where tokens are not refilled automatically, but must be added manually? We have a scenario where we may have a rate limit determined externally and we don't want tokens to automatically come back, we want to force them back in after specific operations are completed. Relative newcomer to bucket4j so pardon if there is an obvious answer to the question that I can't see.

thx

**Comment by vladimir-bukhtoyarov** [comment]
Hello @afm497 

There is no such functionality.

**Reply by afm497** [reply]
Thanks for the quick reply, @vladimir-bukhtoyarov . Is there any way that a Refill policy could be set to 0? In that way as tokens as consumed, they aren;'t refilled until manually added back in? I believe I saw some way possible to add tokens in programmatically, https://javadoc.io/doc/com.github.vladimir-bukhtoyarov/bucket4j-core/latest/io/github/bucket4j/Bucket.html#addTokens-long-
Or does a Refill policy always have to add back tokens?

**Reply by vladimir-bukhtoyarov** [reply]
You can configure very long refill period:
```java 
int capacity = 100;
// Refill can be threaten as effectively zero.
Bandwidth.classic(capacity, Refill.intervally(1, Duration.ofNanos(Long.MAX_VALUE)));
```
